src/metatrain/utils/data/readers/readers.py

```diff
@@ -8,50 +9,75 @@
 from omegaconf import DictConfig
 
 from ..dataset import TargetInfo, TargetInfoDict
-from .systems import SYSTEM_READERS
-from .targets import ENERGY_READERS, FORCES_READERS, STRESS_READERS, VIRIAL_READERS
 
 
 logger = logging.getLogger(__name__)
 
+SUPPORTED_READER_LIBRARIES = ["ase"]
+""":py:class:`list`: list containing all implemented reader libraries"""
+
+READER_FORMATS = {
+    ".xyz": "ase",
+    ".extxyz": "ase",
+}
+""":py:class:`dict`: dictionary mapping file suffixes to default readers"""
+
 
 def _base_reader(
-    readers: dict,
+    target: str,
     filename: str,
-    fileformat: Optional[str] = None,
+    reader: Optional[str] = None,
     dtype: torch.dtype = torch.float32,
     **reader_kwargs,
 ):
-    if fileformat is None:
-        fileformat = Path(filename).suffix
+    if reader is None:
+        try:
+            filesuffix = Path(filename).suffix
+            reader = READER_FORMATS[filesuffix]
+        except KeyError:
+            raise ValueError(
+                f"File suffix {filesuffix!r} is not linked to a default reader "
+                "library. You can reading it by setting a specific 'reader' from the "
+                f"known ones: {', '.join(SUPPORTED_READER_LIBRARIES)} "
+            )
+
+    try:
+        reader_mod = importlib.import_module(
+            name=f".{reader}", package="metatrain.utils.data.readers"
+        )
+    except ImportError:
+        raise ValueError(
+            f"Reader library {reader!r} not supported. Choose from "
+            f"{', '.join(SUPPORTED_READER_LIBRARIES)}"
+        )
 
     try:
-        reader = readers[fileformat]
-    except KeyError:
-        raise ValueError(f"fileformat {fileformat!r} is not supported")
+        reader_met = getattr(reader_mod, f"read_{target}_{reader}")
+    except AttributeError:
+        raise ValueError(f"Reader library {reader!r} can't read {target!r}.")
 
-    return reader(filename, dtype=dtype, **reader_kwargs)
+    return reader_met(filename, dtype=dtype, **reader_kwargs)
 
 
 def read_energy(
     filename: str,
     target_value: str = "energy",
-    fileformat: Optional[str] = None,
+    reader: Optional[str] = None,
     dtype: torch.dtype = torch.float32,
 ) -> List[TensorBlock]:
     """Read energy informations from a file.
 
     :param filename: name of the file to read
     :param target_value: target value key name to be parsed from the file.
-    :param fileformat: format of the system file. If :py:obj:`None` the format is
-        determined from the suffix
+    :param reader: reader library for parsing the file. If :py:obj:`None` the library is
+        tried to determined from the file suffix.
```

Comment:
tried to determined, doesn't sound correct

Reply:
I would just say "the library is determined from the file extension"